signalfx-connection/pom.xml

```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>com.signalfx.public</groupId>
+    <artifactId>clients-parent</artifactId>
+    <version>1.0.30</version>
+  </parent>
+
+  <properties>
+    <!-- This prevents shading from happening in this module -->
+    <skipShaded>true</skipShaded>
+    <maven.deploy.skip>false</maven.deploy.skip>
+  </properties>
+
+  <artifactId>signalfx-connection</artifactId>
+  <name>SignalFx connection</name>
+  <description>Shared SignalFx connection related classes</description>
+  <url>https://www.splunk.com</url>
+
+  <licenses>
+    <license>
+      <name>Apache License 2.0</name>
+      <url>https://www.apache.org/licenses/LICENSE-2.0.html</url>
+      <distribution>repo</distribution>
+    </license>
+  </licenses>
+
+  <scm>
+    <connection>scm:git:git@github.com:signalfx/signalfx-java.git</connection>
+    <developerConnection>scm:git:git@github.com:signalfx/signalfx-java.git</developerConnection>
+    <url>git@github.com:signalfx/signalfx-java.git</url>
+  </scm>
+
+  <developers>
+    <developer>
+      <id>signalfx</id>
+      <name>SignalFx</name>
+      <email>support+java@signalfx.com</email>
+      <organization>SignalFx, Inc</organization>
+      <organizationUrl>http://www.signalfx.com</organizationUrl>
+    </developer>
+  </developers>
+
+  <dependencies>
+    <dependency>
+      <groupId>com.signalfx.public</groupId>
+      <artifactId>signalfx-endpoint</artifactId>
+      <version>${version}</version>
+    </dependency>
+    <dependency>
+      <groupId>commons-io</groupId>
+      <artifactId>commons-io</artifactId>
+    </dependency>
+    <!-- test -->
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.httpcomponents</groupId>
+      <artifactId>httpclient</artifactId>
+    </dependency>
```

Comment:
That's defined as a compile dependency in the parent pom

Reply:
I think the parent just sets up the versions...so this is definitely required, I confirmed that removing it breaks the build.

Reply:
> I think the parent just sets up the versions...so this is definitely required, I confirmed that removing it breaks the build.

I'm not saying that it's not required; I'm saying that if it's a test dep it needs `<scope>test</scope>`

Reply:
Yeah, that doesn't work because it's used by the `RetryStrategy` and `RetryHandler` which are production classes.